pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -229,11 +229,17 @@ public static ColumnMetadataImpl fromPropertiesConfiguration(String column, Prop
     if (defaultNullValueString != null && storedType == DataType.STRING) {
       defaultNullValueString = CommonsConfigurationUtils.recoverSpecialCharacterInPropertyValue(defaultNullValueString);
     }
+    int maxLength = config.getInt(Column.getKeyFor(column, Column.SCHEMA_MAX_LENGTH), FieldSpec.DEFAULT_MAX_LENGTH);
+    String maxLengthExceedStrategyString =
+        config.getString(Column.getKeyFor(column, Column.SCHEMA_MAX_LENGTH_EXCEED_STRATEGY), null);
+    FieldSpec.MaxLengthExceedStrategy maxLengthExceedStrategy = maxLengthExceedStrategyString != null
+        ? FieldSpec.MaxLengthExceedStrategy.valueOf(maxLengthExceedStrategyString) : null;
     FieldSpec fieldSpec;
     switch (fieldType) {
       case DIMENSION:
         boolean isSingleValue = config.getBoolean(Column.getKeyFor(column, Column.IS_SINGLE_VALUED));
-        fieldSpec = new DimensionFieldSpec(column, dataType, isSingleValue, defaultNullValueString);
+        fieldSpec = new DimensionFieldSpec(column, dataType, isSingleValue, maxLength,
+            defaultNullValueString, maxLengthExceedStrategy);
         break;
       case METRIC:
         fieldSpec = new MetricFieldSpec(column, dataType, defaultNullValueString);
```

Comment:
This also applies to metric columns

Reply:
I see the constructor was missing in MetricFieldSpec so added that as well